lldb/source/Plugins/SymbolFile/DWARF/DWARFASTParserClang.cpp

```diff
@@ -1595,49 +1627,67 @@ DWARFASTParserClang::GetCPlusPlusQualifiedName(const DWARFDIE &die) {
   if (qualified_name.empty())
     qualified_name.append("::");
 
-  qualified_name.append(name);
+  qualified_name.append(unique_typename.GetCString());
   qualified_name.append(GetDIEClassTemplateParams(die));
 
-  return qualified_name;
+  unique_typename = ConstString(qualified_name);
```

Comment:
Nit: I'd find it slightly clearer if we returned the new name like the old function did (probably don't even need the parameter to be a `ConstString`)

Reply:
This functions modifies both `unique_typename` and `decl_declaration` if it's c++, so there're two out values. Returning one out value while modifying another out value via reference parameter looks strange.